### ISSUE TITLE
Price per block could increase

### ISSUE THREAD

**Issue by That3Percent** [edgeandnode/subscription-payments#2]
Here:
https://github.com/edgeandnode/subscription-payments-contract/blob/7b0368e4da25ece37440e0999fbeee652ad8b103/contracts/Subscriptions.sol#L90

there is no "defense" to the consumer for the price per block to increase suddenly. This could be used by a malicious entity to drain funds

**Comment by Theodus** [comment]
My understanding is that `transferFrom` requires the sender to first `approve` the transfer of up to some amount of tokens. The expectation is that the subscriber would only approve the expected amount so an unexpected increase would result in the transaction being reverted.

**Comment by That3Percent** [comment]
True, but a frequent pattern is to approve an unlimited amount if they expect to repeatedly interact with the contract. Something we should run by experts when the time comes.
